### PR TITLE
e2e: Add web3signer component

### PR DESCRIPTION
**Pull request by prestonvanloon**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

Part of web3signer client implementation (#9994).

This PR introduces Consensys' web3signer as a component in e2e.
The web3signer component is configured to load the keys from the deterministic interop genesis key generator. I've added a local component test to ensure that the web3signer starts and returns the appropriate amount of keys. The validator component has been wired to be configured with the web3signer. Once other components of the web3signer client implementation are complete, then the test may be unskipped with the TODOs resolved.

**Which issues(s) does this PR fix?**

**Other notes for review**

### CODE REVIEW

File: testing/endtoend/components/web3remotesigner.go [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+	if len(keys) == 0 {
+		return nil, errors.New("no keys returned")
+	}
+
+	var pks []bls.PublicKey
+	for _, key := range keys {
+		if ctx.Err() != nil {
+			return nil, ctx.Err()
+		}
+		raw, err := hex.DecodeString(strings.TrimPrefix(key, "0x"))
```
Comment (james-prysm) [inline testing/endtoend/components/web3remotesigner.go:180]: I think hexutil library we use can do the prefix and the trimming

File: testing/endtoend/components/web3remotesigner.go [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+		return err
+	}
+	for i, pk := range priv {
+		if ctx.Err() != nil {
+			return ctx.Err()
+		}
+		rkf := &rawKeyFile{
+			Type:       "file-raw",
+			KeyType:    "BLS",
+			PrivateKey: "0x" + hex.EncodeToString(pk.Marshal()),
```
Comment (james-prysm) [inline testing/endtoend/components/web3remotesigner.go:214]: also related to the previous hexutil comment

File: testing/endtoend/components/validator.go [hunk 3]
```diff
@@ -148,6 +148,27 @@ func (v *ValidatorNode) Start(ctx context.Context) error {
...
+		args = append(args, fmt.Sprintf("--%s=localhost:%d", "validators-external-signer-url", Web3RemoteSignerPort))
+		// Write the pubkeys as comma seperated hex strings with 0x prefix.
+		// See: https://docs.teku.consensys.net/en/latest/HowTo/External-Signer/Use-External-Signer/
+		_, pubs, err := interop.DeterministicallyGenerateKeys(uint64(offset), uint64(validatorNum))
+		if err != nil {
+			return err
+		}
+		var hexPubs []string
+		for _, pub := range pubs {
+			hexPubs = append(hexPubs, "0x"+hex.EncodeToString(pub.Marshal()))
```
Comment (james-prysm) [inline testing/endtoend/components/validator.go:162]: also related to hexutil comment

**Review by james-prysm — APPROVED** [review]
my items of concern were addressed for this e2e, will need to update this after [this pr](https://github.com/prysmaticlabs/prysm/pull/10056) is ready

**Review by rauljordan — APPROVED** [review]

**Pull request merged**